app/models/notification.rb

```diff
@@ -26,58 +26,79 @@ class Notification < ApplicationRecord
 
   paginates_per 20
 
+  API_ATTRIBUTE_MAP = {
+    repository_id: [:repository, :id],
+    repository_full_name: [:repository, :full_name],
+    repository_owner_name: [:repository, :owner, :login],
+    subject_title: [:subject, :title],
+    subject_type: [:subject, :type],
+    subject_url: [:subject, :url],
+    reason: [:reason],
+    unread: [:unread],
+    updated_at: [:updated_at],
+    last_read_at: [:last_read_at],
+    url: [:url],
+    github_id: [:id]
+  }.freeze
```

Comment:
I thought having a single map would be more maintainable than coding the paths in the update methods